MML Issue 927: Add LAM FuelTank MiscType

SJuliez opened the pull request:
This adds LAM Fuel Tanks of IO, p.114

@HammerGS : Please review the data. I did make a slight change to the LAM Bomb Bays from IO, pp.220 and 221. I wasnt sure about the production dates as IO doesnt list the Bomb and FTs in the Grand Table of Tech. Id say we can't err on too much availability as both are 100% restricted to LAMs. Otherwise this only adds a helper method to Entity that is used for the MML side of this.

HammerGS reviewed (commented):
What I'd do for the fuel tanks is mirror the LAM availability.

.setISAdvancement(2680, 2684, DATE_NONE, 3085)

.setClanAdvancement(DATE_NONE, 2684, DATE_NONE, 2825)

That would imply that came into being and progressed the same time as LAM's progressed.

HammerGS commented on megamek/src/megamek/common/MiscType.java:
```diff
@@ -11173,19 +11174,36 @@ public static MiscType createLightMinesweeper() {
+        misc.rulesRefs = "114, IO";
+        // IO, p.220/221:
+        misc.techAdvancement.setTechBase(TECH_BASE_ALL).setAdvancement(DATE_ES, DATE_ES)
+                .setTechRating(RATING_B).setAvailability(RATING_B, RATING_A, RATING_A, RATING_A)
```
.setISAdvancement(2680, 2684, DATE_NONE, 3085).setClanAdvancement(DATE_NONE, 2684, DATE_NONE, 2825)

SJuliez replied:
There is no difference here, right? I can't spot any

SJuliez replied:
Got confused with the comments here. This one is the Fuel tank :)

HammerGS commented on megamek/src/megamek/common/MiscType.java:
```diff
@@ -11173,19 +11174,36 @@ public static MiscType createLightMinesweeper() {
             .setPrototypeFactions(F_TH).setProductionFactions(F_TH)
-            .setTechRating(RATING_D).setAvailability(RATING_D, RATING_E, RATING_F, RATING_F)
-            .setStaticTechLevel(SimpleTechLevel.EXPERIMENTAL);
+            .setTechRating(RATING_D).setAvailability(RATING_B, RATING_E, RATING_E, RATING_E)
```
.setISAdvancement(2680, 2684, DATE_NONE, 3085).setClanAdvancement(DATE_NONE, 2684, DATE_NONE, 2825)

SJuliez replied:
Don't wish to appear argumentative over a non-issue, but doesn't this contradict IO p.221 directly? (These fuel tanks aren't available for anything but LAMs so they are tied to their availability anyway. I suppose the authors of IO thought a Fuel Tank's just a Fuel Tank)

Windchild292 reviewed (approved):
The code looks good, I'll leave the data to Hammer

SJuliez commented:
@HammerGS ok I corrected the tech advancement. Please take a look.

SJuliez commented:
@HammerGS I notice that Bomb Bays go extinct by 3085 but I can still constrcut a LAM in 3145. Meaning I can't give that LAM Bomb Bays. Not sure what's supposed to happen. Should MML forbid LAM construction in 3145?

HammerGS commented:
> @HammerGS ok I corrected the tech advancement. Please take a look.

Looks good.

> @HammerGS I notice that Bomb Bays go extinct by 3085 but I can still constrcut a LAM in 3145. Meaning I can't give that LAM Bomb Bays. Not sure what's supposed to happen. Should MML forbid LAM construction in 3145?

According to IntOps LAMs return in 3085 and stay available for construction. Of course doesn't mean anyone's building them. For the Bomb Bays lets remove the extinct in 3085 to match the LAM availability.

SJuliez commented:
@HammerGS OK as discussed I reset the Fuel tank dates to ES, ES and removed the extinct date from the Bomb Bays to keep them available to LAMs after 3085. I also removed that equipment that's been commented out since 2017.

HammerGS reviewed (approved):
I’m good with the data stuff.

Windchild292 commented:
The merge conflict is just whitespace, so I'm going to fix it, wait for CI, and then merge it.
